### DISCUSSION TITLE
Support Z2T Hue Emulation for Zigbee lights with multiple endpoints

### DISCUSSION THREAD

**Discussion by urechi** [arendst/Tasmota#16508] [Ideas]
Hi all,

I believe at the moment Alexa can discover only endpoint 1 of the Zigbee light switch. It would be nice to support light switches with multiple endpoints. Thanks @s-hadinger for writing the code to make this possible and also many thanks to all the contributors of Tasmota.

**Comment by urechi** [comment]
I'll add some more details about the above issue. I have [this](https://www.amazon.co.uk/gp/product/B09B9TZYH6/?th=1) Zigbee light switch with 2 gang.

```
16:55:00.425 CMD: ZBStatus3 0xD4AF
16:55:00.426 SRC: WebConsole from 192.168.1.44
16:55:00.428 CMD: Grp 0, Cmd 'ZBSTATUS', Idx 3, Len 6, Pld 54447, Data '0xD4AF'
16:55:00.442 MQT: stat/SmartHub-D3F00C/RESULT = {"ZbStatus3":[{"Device":"0xD4AF","Name":"Kitchen Light Switch","IEEEAddr":"0xEC1BBDFFFEC72CBA","ModelId":"TS0002","Manufacturer":"_TYZB01_wmak4qjy","Endpoints":[1,2],"Config":["L00.0","O01","O02"],"Power":1,"Power2":1,"Reachable":true,"LastSeen":6,"LastSeenEpoch":1663433694,"LinkQuality":63}]}
```

I have enabled Hue Emulation and run `ZBLight` with simple On/Off light as per documentation [here](https://tasmota.github.io/docs/Zigbee/#zigbee-and-hue-emulation-for-alexa).
```
16:56:44.164 CMD: ZBLight 0xD4AF,0
16:56:44.165 SRC: WebConsole from 192.168.1.44
16:56:44.167 CMD: Grp 0, Cmd 'ZBLIGHT', Idx 1, Len 8, Pld 54447, Data '0xD4AF,0'
16:56:44.177 MQT: tele/SmartHub-D3F00C/D4AF/SENSOR = {"0xD4AF":{"Device":"0xD4AF","Name":"Kitchen Light Switch","Reachable":true,"Power":1,"Light":0,"BatteryPercentage":146}} (retained)
16:56:44.193 MQT: stat/SmartHub-D3F00C/RESULT = {"ZbLight":"Done"}
```

Alexa, discover devices:
I've found `Kitchen Light Switch` 
Only one endpoint is detected. I was expecting Alexa to discover one light with two channels/endpoints or maybe two lights.
I am not sure if this is a bug/limitation or if I do something wrong. Any thoughts?


![IMG_3856](https://user-images.githubusercontent.com/38536263/190868261-c4ea2eb8-d721-4509-9232-51fe3b5075a1.PNG)

**Comment by s-hadinger** [comment]
This is a current limitation: only one Alexa device is exposed per zigbee device. For bulbs it was enough but I didn't think about switches. I will see how to add support for multiple endpoints. I don't see why it wouldn't work as well but there are some details to sort out.

**Reply by urechi** [reply]
Thanks for getting back. Things might get a bit complicated when there is a setup like I have, for example, one of my light switches with 3 gangs named `Bathroom Light Switch`:
- Endpoint 1 controls the bathroom light
- Endpoint 2 controls the bathroom fan
- Endpoint 3 controls the landing light

Different names for each endpoint?
Does Hue API/Alexa support entities other than lights?


<img width="531" alt="Screenshot 2022-09-19 at 11 52 54" src="https://user-images.githubusercontent.com/38536263/191002688-a366ce46-a2a1-48c0-8347-47363301df26.png">
Also, a remark about the current Tasmota web UI, only the endpoint 1 state is displayed. Maybe it makes sense to display a list of states based on the number of endpoints available for that device? For a 2 gang switch it could display the states like:

<img width="531" alt="Screenshot 2022-09-19 at 12 02 28" src="https://user-images.githubusercontent.com/38536263/191003828-1f89b481-9d90-4591-8f93-1f8db573a9a2.png">

**Comment by s-hadinger** [comment]
@urechi This is now supported, please see https://github.com/arendst/Tasmota/pull/16718
